validator/client/iface/validator.go

```diff
@@ -57,4 +57,5 @@ type Validator interface {
 	ReceiveBlocks(ctx context.Context, connectionErrorChannel chan<- error)
 	HandleKeyReload(ctx context.Context, newKeys [][fieldparams.BLSPubkeyLength]byte) (bool, error)
 	CheckDoppelGanger(ctx context.Context) error
+	PrepareBeaconProposer(ctx context.Context, km keymanager.IKeymanager) error
```

Comment:
From the interface, this feels like it is about preparing proposer duties, which is unclear. We should probably rename this to prepare fee recipients